topic/machine-learning/llm-langchain/cratedb_rag_customer_support_langchain.ipynb

```diff
@@ -410,11 +460,160 @@
     "id": "aQnmpCIZa13L",
     "outputId": "a10c71a7-a6c7-4f83-c069-df49703d4654"
    },
-   "outputs": [],
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'To update the shipping address on an existing Microsoft Store order, you can contact Microsoft support either by calling them or reaching out through email. They will assist you in updating the shipping address for your order.'"
+      ]
+     },
+     "execution_count": 48,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
    "source": [
     "chat_completion.choices[0].message.content"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## CrateDB &#129505; OpenSource\n",
+    "\n",
+    "In the second part of this notebook we build a Retrieval-Augmented Generation (RAG) system using CrateDB with Jina AI and the Llama open-source models. \n",
+    "\n",
+    "### Configure Jina AI API "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Jina AI embeddings on Hugging Face are a suite of open-source models designed for encoding text data into high-dimensional vectors, which can be used in various natural language processing tasks such as semantic search, text classification, and clustering. These embeddings utilize the BERT architecture and are specially tuned for longer sequence lengths up to 8192 tokens, which is notably higher than the standard BERT model's capacity.\n",
+    "\n",
+    "To use Jina AI embeddings you need a Jina AI API key which can be easily obtained once you sign up for an account on the Jina AI website."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_community.embeddings import JinaEmbeddings\n",
+    "\n",
+    "getenvpass(\"JINA_API_KEY\", prompt=\"Jina API key:\")\n",
+    "embeddings = JinaEmbeddings()"
```

Comment:
Please note: On CI, when not providing the `JINA_API_KEY` to its environment, the test run will exit here using the SKIP outcome, so that it will not fail the software test.

While this is intended for convenience purposes, however, subsequent code will not be tested, and it should be taken into consideration.

Reply:
Also: Would using LC's `JinaEmbeddings` need other dependencies to be installed, in order to work properly? If so, let's also add them to the `requirements.txt` file?

Reply:
No other dependencies are needed. `JinaEmbeddings` is part of the langchain community module.